tests/unit/s2n_disable_validity_period_validation_test.c

```diff
@@ -0,0 +1,251 @@
+                .expected_error = S2N_ERR_OK,
+            },
+
+            /* Validation should fail for an expired certificate. */
+            {
+                .cert_pem_path = S2N_EXPIRED_CERT_CHAIN,
+                .key_pem_path = S2N_EXPIRED_KEY,
+                .disable_validity_period_validation = false,
+                .expected_error = S2N_ERR_CERT_EXPIRED,
+            },
+
+            /* Validation should succeed for an expired certificate when validity period validation
+             * is disabled.
+             */
+            {
+                .cert_pem_path = S2N_EXPIRED_CERT_CHAIN,
+                .key_pem_path = S2N_EXPIRED_KEY,
+                .disable_validity_period_validation = true,
+                .expected_error = S2N_ERR_OK,
+            },
+        };
+        /* clang-format on */
+
+        /* s2n_x509_validator test */
+        for (int i = 0; i < s2n_array_len(test_cases); i++) {
+            DEFER_CLEANUP(struct s2n_x509_trust_store trust_store = { 0 }, s2n_x509_trust_store_wipe);
+            s2n_x509_trust_store_init_empty(&trust_store);
+
+            char cert_chain[S2N_MAX_TEST_PEM_SIZE] = { 0 };
+            EXPECT_SUCCESS(s2n_read_test_pem(test_cases[i].cert_pem_path, cert_chain, S2N_MAX_TEST_PEM_SIZE));
+            EXPECT_SUCCESS(s2n_x509_trust_store_add_pem(&trust_store, cert_chain));
+
+            DEFER_CLEANUP(struct s2n_x509_validator validator, s2n_x509_validator_wipe);
+            EXPECT_SUCCESS(s2n_x509_validator_init(&validator, &trust_store, 0));
+
+            DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+            EXPECT_NOT_NULL(config);
+            EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "default"));
+
+            if (test_cases[i].disable_validity_period_validation) {
+                EXPECT_SUCCESS(s2n_config_disable_x509_validity_period_validation(config));
+            }
+
+            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT), s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(conn);
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+            EXPECT_SUCCESS(s2n_set_server_name(conn, "localhost"));
+
+            DEFER_CLEANUP(struct s2n_stuffer cert_chain_stuffer = { 0 }, s2n_stuffer_free);
+            EXPECT_OK(s2n_test_cert_chain_data_from_pem(conn, test_cases[i].cert_pem_path, &cert_chain_stuffer));
+            uint32_t chain_len = s2n_stuffer_data_available(&cert_chain_stuffer);
+            uint8_t *chain_data = s2n_stuffer_raw_read(&cert_chain_stuffer, chain_len);
+            EXPECT_NOT_NULL(chain_data);
+
+            DEFER_CLEANUP(struct s2n_pkey public_key_out = { 0 }, s2n_pkey_free);
+            EXPECT_SUCCESS(s2n_pkey_zero_init(&public_key_out));
+            s2n_pkey_type pkey_type = S2N_PKEY_TYPE_UNKNOWN;
+
+            s2n_error expected_error = test_cases[i].expected_error;
+            if (expected_error == S2N_ERR_OK) {
```

Comment:
Can't you do:
```suggestion
            EXPECT_EQUAL(s2n_x509_validator_validate_cert_chain(&validator, conn, chain_data, chain_len, &pkey_type, &public_key_out), test_cases[i].expected_error)
```
Honestly I don't know if this will work but it's cleaner.

Reply:
I'm not sure this works since the return value needs to be checked before checking the error, which is what the different macros do. However, I tried to make this cleaner by not duplicating the function that's being tested.